toolkit/tools/imagecustomizerapi/overlay.go

```diff
@@ -5,69 +5,86 @@ package imagecustomizerapi
-	UpperDir  string               `yaml:"upperDir"`
-	WorkDir   string               `yaml:"workDir"`
-	Partition *IdentifiedPartition `yaml:"partition"`
+	LowerDirs         []string `yaml:"lowerDirs"`
+	UpperDir          string   `yaml:"upperDir"`
+	WorkDir           string   `yaml:"workDir"`
+	MountPoint        string   `yaml:"mountPoint"`
+	IsRootfsOverlay   bool     `yaml:"isRootfsOverlay"`
+	MountDependencies []string `yaml:"mountDependencies"`
+	MountOptions      string   `yaml:"mountOptions"`
 }
 
 func (o *Overlay) IsValid() error {
-	// Validate paths for UpperDir, WorkDir, and LowerDir
+	// Validate paths for LowerDirs, UpperDir, WorkDir, and MountPoint.
+	for _, lowerDir := range o.LowerDirs {
+		if err := validatePath(lowerDir); err != nil {
+			return fmt.Errorf("invalid lowerDir (%s):\n%w", lowerDir, err)
+		}
+	}
 	if err := validatePath(o.UpperDir); err != nil {
 		return fmt.Errorf("invalid upperDir (%s):\n%w", o.UpperDir, err)
 	}
 	if err := validatePath(o.WorkDir); err != nil {
 		return fmt.Errorf("invalid workDir (%s):\n%w", o.WorkDir, err)
 	}
-	if err := validatePath(o.LowerDir); err != nil {
-		return fmt.Errorf("invalid lowerDir (%s):\n%w", o.LowerDir, err)
+	if err := validatePath(o.MountPoint); err != nil {
+		return fmt.Errorf("invalid mountPoint (%s):\n%w", o.MountPoint, err)
+	}
+	for _, dependency := range o.MountDependencies {
+		if err := validatePath(dependency); err != nil {
+			return fmt.Errorf("invalid mountDependencies (%s):\n%w", dependency, err)
+		}
+	}
+
+	if strings.Contains(o.MountOptions, " ") {
+		return fmt.Errorf("mountOptions (%s) contains spaces and is invalid", o.MountOptions)
 	}
 
-	// Check if UpperDir and WorkDir are identical
+	// Check if UpperDir and WorkDir are identical.
 	if o.UpperDir == o.WorkDir {
 		return fmt.Errorf("upperDir and workDir must be distinct, but both are '%s'", o.UpperDir)
 	}
 
-	// Check if UpperDir is a subdirectory of WorkDir or vice versa
+	// Check if UpperDir is a subdirectory of WorkDir or vice versa.
 	if isSubDirString(o.UpperDir, o.WorkDir) {
 		return fmt.Errorf("upperDir (%s) should not be a subdirectory of workDir (%s)", o.UpperDir, o.WorkDir)
 	}
 	if isSubDirString(o.WorkDir, o.UpperDir) {
 		return fmt.Errorf("workDir (%s) should not be a subdirectory of upperDir (%s)", o.WorkDir, o.UpperDir)
 	}
 
-	if o.Partition != nil {
-		if err := o.Partition.IsValid(); err != nil {
-			return fmt.Errorf("invalid partition:\n%w", err)
-		}
-	}
-
 	return nil
 }
 
-func validatePath(path string) error {
-	// Check if the path is empty
-	if path == "" {
+func validatePath(filePath string) error {
+	// Check if the path is empty.
+	if filePath == "" {
 		return fmt.Errorf("path cannot be empty")
 	}
 
-	// Check if the path contains spaces
-	if strings.Contains(path, " ") {
-		return fmt.Errorf("path (%s) contains spaces and is invalid", path)
+	// Check if the path contains spaces.
+	if strings.Contains(filePath, " ") {
```

Comment:
It would probably be good to also exclude tabs, newlines, colons, and commas to help avoid the path values from messing up the fstab file's formatting.

[strings.ContainsAny](https://pkg.go.dev/strings#ContainsAny) could be handy for this.

Also, if you could also use `validatePath` for `MountPoint.Path` that would be greatly appreciated.